### DISCUSSION TITLE
Q: Can we use OrchardCore for purposes of standing up an auth server

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore#7207] [Q&A]
Hello,

First, great body of work here, I think. I may eventually adopt the full gamut CMS front to back. Eventually. And forgive me, I may be barking way up the wrong tree with all this, so bear with me. Or perhaps someone can point me in a better direction.

But for now, I am investigating options concerning a family potentially related or unrelated SPA verticals, and I'd like to possibly leverage something like an OrchardCore for purposes of hosting our auth and ClaimsPrinciple identity concerns.

I have a loose comprehension of the whole auth, ClaimsPrinciple workflow. My understanding is, the last thing I really want to do is be that intimately familiar with any of the upstream identities themselves. The most I may need to keep track of are, perhaps, bearer tokens, and connect them with internal user accounts, then align my SPA app claims with those token accounts. Along these lines.

Segue, enter, possibly, OrchardCore. I understand it offers basic admin panels, this sort of thing, for these kind of effort? Again, if this is incorrect, barking up the wrong tree, please forgive me.

Thanks!

**Comment by mwpowellhtx** [comment]
And to clarify, specifically, I think it was [OpenId](https://docs.orchardcore.net/en/dev/docs/reference/modules/OpenId/) support that piqued my curiosity as being one possible solution.

**Comment by jtkech** [comment]
Not an expert in security, @kevinchalet is the expert that can confirm, but almost everything is possible with the OC.OpenId module, including configuring an auth server, moreover, because an OrchardCore app can have multiple tenants / sites, you can configure multiple auth servers, and any group of tenants can be the clients of a server implemented by another tenant.

**Comment by kevinchalet** [comment]
@jtkech you're absolutely right! 👏

**Comment by sebastienros** [comment]
We need a recipe for that scenario.
